server/server.go

```diff
@@ -381,20 +352,27 @@ func NewServer(userConfig UserConfig, config Config) (*Server, error) {
 		EnableDiffMarkdownFormat: userConfig.EnableDiffMarkdownFormat,
 	}
 
-	boltdb, err := db.New(userConfig.DataDir)
+	// moved
+	boltdb, err := db.NewBoltdb(userConfig)
 	if err != nil {
 		return nil, err
 	}
 	var lockingClient locking.Locker
 	var applyLockingClient locking.ApplyLocker
+
+	// moved
 	if userConfig.DisableRepoLocking {
 		lockingClient = locking.NewNoOpLocker()
 	} else {
 		lockingClient = locking.NewClient(boltdb)
 	}
+
 	applyLockingClient = locking.NewApplyClient(boltdb, userConfig.DisableApply)
+
+	// moved
 	workingDirLocker := events.NewDefaultWorkingDirLocker()
 
+	// moved
 	var workingDir events.WorkingDir = &events.FileWorkspace{
```

Comment:
Can we use the provider here instead? https://github.com/lyft/atlantis/blob/080fc925c63ecfdf4b39fbe3f6a82229df62025d/server/events/working_dir.go#L72-L76 
Even though I'm guessing we'll be removing it when we move everything to wire!

Reply:
Yeah i think i added this when i was just experimenting.